bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+# Blockchain
+
+## Block
+
+Block is a bunch of transactions that have been added to blockchain. A bitmark block contains data as follows:
+
+  size (bytes)    data
+ --------------  ----------------------------------------------------------------
+   2              block version number
+   2              transaction count inside the block
+   8              block number (block height)
+  32              argon2 hash of previous block header
+  32              sha3 hash of all transactions in the block (merkle root)
+   8              timestamp of block generation time, since 1970-01-01T00:00 UTC
+   8              current difficulty in compact mode
+   8              number to make hash meets difficulty
+   ... to end     transactions
+
+Different types of transaction include different length and content inside.
+
+### Block Verification
+
+When a block is received, the following checks will be made:
+
+1. Length of all parts inside a block should match
+1. Incoming block version number should be equal or largen than current block version number
+1. Incoming block height is 1 ahead of current block height
+1. Incoming block contans at least 1 transaction, at most 9999 transactions inside
+1. Incoming block generation time, compared to received time, should within 5 minutes
+1. Incoming block generation time can not be larger then 10 minutes compares to current block generation time
+1. Incoming block generation time within specific range
+1. Incoming block difficulty is same as current difficulty
+1. Incoming block hash meets difficulty criteria
+1. Incoming block's previous block hash same as current block hash
+1. Incoming transaction fits each type's format and length
+1. Incoming transaction owner ownership is valid
+1. Incoming transaction payment is valid
+1. Incoming block's sha3 of all transactions inside is valid
+
+After a block is verified, a node stores that block into its internal
+database (leveldb), then node broadcast this newly saved block to all connected
+nodes. When other node receives the broadcast of a block, each node
+verifies block correctness by rules described above.
+
+### Block Synchronization
+
+Bitmark blockchain uses both proof-of-work (PoW) and majority votes to
+choose blocks.
+
+#### Proof of Work
+
+A proof of work is a piece of data which is difficult (costly, time-consuming) to produce but easy for others to verify and which satisfies certain requirements. Producing a proof of work can be a random process with low probability so that a lot of trial and error is required on average before a valid proof of work is generated. Bitcoin uses the Hashcash proof of work system.
```

Comment:
If we're going to pull content from the Bitcoin Wiki (or elsewhere) we should be certain to reference it. 

Unsurprisingly, I like my definition of Proof of Work better, from one of our Medium articles:

Proof of work is the traditional method of Sybil defense on a blockchain, used by Bitcoin and (at the moment) by Ethereum. Every participant is given the opportunity to solve a math problem. Whoever manages to do so is allowed to propose a block for the blockchain, contributing to the consensus and the growth of the ledger. Sybil defense is provided by the fact that it’s very hard to make these calculations, and so has real costs in energy.

https://medium.com/clean-titles/a-philosophy-of-blockchain-what-would-satoshi-nakamoto-think-of-proof-of-stake-f9aac8e4d005

Reply:
Sure, no problem, updated